openpype/hosts/maya/api/pipeline.py

```diff
@@ -211,6 +246,18 @@ def _on_scene_open(*args):
     emit_event("open")
 
 
+def _check_lock_file(*args):
+    emit_event("check.lock")
```

Comment:
If you're retriggering event based on different event then use proper callback name for event source, not for usage.
```suggestion
def _after_scene_open(*args):
    emit_event("after.file.open")
```

Comment n2: I don't think you have to retrigger events in these callbacks. You can just call what is in `on_check_lock` and remove `on_check_lock` (it's not used elsewhere).

The same for `_before_close_maya` and `_before_scene_open`. All of them are event callbacks that are triggerering new events with callbacks only in this file.

Reply:
I need to test for the necessity with _before_scene_open and the related callback and I think I can put on_check_lock on the function which already uses the kAfterOpen callback.
I am not quite sure about _before_close_maya as it is related to kMayaExiting